File: eolymp/map/dino_and_array.py
import collections


d = {}

for _ in range(int(input())):
    k, v = [int(i) for i in input().split()]
    d[k] = v

k = int(input())

od = collections.OrderedDict(sorted(d.items()))

idx = 1

for key, v in od.items():
    if(idx+v >= k):
        print(key)
        exit(0)
    else:
        idx+=v

File: eolymp/map/dino_and_array.cpp
#include<map>
#include<iostream>

using namespace std;

int main(){
    map<long long, long long> m;
    long long n;
    long long a, b;
    cin >> n;
    while(n--){
        cin >> a >> b;
        m[b] += a;
    }
    long long k;
    cin >> k;
    long long idx = 0;
    for(auto p : m){
        idx += p.second;
        if(idx >= k){
            cout << p.first;
            exit(0);
        }
    }
    return 0;
}
